app.js

```diff
@@ -0,0 +1,780 @@
+  }
+  if(randomNum == 2) {
+    setBossHealth(players["boss"].health - 5);
+    displayDamage(5);
+    return 5;
+  }
+  else {
+    setBossHealth(players["boss"].health - 10);
+    displayDamage(10);
+    return 10;
+  }
+}
+
+function grow() {
+  setEnergy(players["player"].energy - 2);
+  if((players["player"].health + (3*players["player"].cardsPlayed)) < 50) {
+    setPlayerHealth(players["player"].health + (3*players["player"].cardsPlayed));
+  } else {
+    setPlayerHealth(50);
+  }
+}
+function spirit() {
+  setEnergy(players["player"].energy+1);
+}
+
+function cleanse() {
+  players["player"].status = [];
+  setProperty("player_status","text", "");
+}
+
+function club() {
+  for(var i=0; i<hand.length; i++) {
+    if(hand[i].name != "none") {
+      return;
+    }
+  }
+  setBossHealth(players["boss"].health - 10);
+  displayDamage(10);
+  return 10;
+}
+
+function fortify() {
+  if(players["player"].shield >= 5) {
+    setPlayerShield(players["player"].shield + 5);
+  }
+}
+
+//---------------------------------------------------------------------------------
+
+function bossTurn() {
+  isPlayerTurn = false;
+  if(checkLifePoints()) {
+    return;
+  }
+  setBossShield(0);
+  //pick and execute an attack
+  var bossEvent = bossAttacks[randomNumber(0,bossAttacks.length-1)];
+  setTimeout(function() {
+    setProperty("boss_description","text","Time Keeper is using " + bossEvent.name);
+  }, 1000);
+  setTimeout(function() {
+    window[bossEvent.name]();
+  },3000);
+  setTimeout(function() {
+    setProperty("boss_description","text", "");
+  }, 4000);
+  setTimeout(function() {
+    useStatus();
+    startTurn();
+  }, 5000);
+  setTimeout(function() {
+    
+  }, 5000);
+}
+
+//Boss cards
+function strike() {
+  setPlayerHealth(players["player"].health - 15);
+  displayDamage(15);
+}
+function slash() {
+  players["player"].status.push(JSON.parse(JSON.stringify(statusEffects.fire)));
+  setPlayerHealth(players["player"].health-10);
+  showStatus();
+  displayDamage(10);
+}
+
+function reinforce() {
+  setBossShield(players["boss"].shield + 10);
+  setPlayerHealth(players["player"].health - 10);
+  displayDamage(10);
+}
+
+//-----------------------------------------------------------------------------
+
+
+
+//Deck creation
+
+//data split for each "screen"
+var showCase = [
+  [cards.spirit,cards.club,cards.fortify,cards.cleanse,cards.blankCard], 
+  [cards.smite, cards.block, cards.draw,cards.heal,cards.flame], 
+  [cards.smear,cards.dash,cards.chain,cards.fortune,cards.grow]
+];
+
+//starting idx
+var showCaseIdx = 1;
+
+
+function switchShowCase(button) {
+  if (button == "right") {
+    if ((showCaseIdx + 1) > 2) {
+      showCaseIdx = 0;
+    }
+    else {
+      showCaseIdx++;
+    }
+  }
+  if (button == "left") {
+    if((showCaseIdx - 1) < 0) {
+      showCaseIdx = 2;
+    }
+    else {
+      showCaseIdx--;
+    }
+  }
+  for(var i=0; i<5; i++) {
+    setProperty("sample_"+i,"image", showCase[showCaseIdx][i].image);
```

Comment:
I'd go ahead and make this a little wordier to make it clearer

```
for(var i=0; i<numCardsShown; i++) {
  var newCard = lookupCard(i)
  setProperty("showcase_"+i,"image", newCard.image);
}
```

Obviously you'd need to change the image IDs to `showcase_` as well. `sample_` probably doesn't make sense as an ID anymore. And you can set the `numCardsShown` as a global.